src/rebar_prv_plugins.erl

```diff
@@ -0,0 +1,74 @@
+-module(rebar_prv_plugins).
+
+-behaviour(provider).
+
+-export([init/1,
+         do/1,
+         format_error/1]).
+
+-include("rebar.hrl").
+-include_lib("providers/include/providers.hrl").
+
+-define(PROVIDER, list).
+-define(NAMESPACE, plugins).
+-define(DEPS, []).
+
+-spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
+init(State) ->
+    State1 = rebar_state:add_provider(
+            State,
+            providers:create([
+                    {name, ?PROVIDER},
+                    {module, ?MODULE},
+                    {namespace, ?NAMESPACE},
+                    {bare, true},
+                    {deps, ?DEPS},
+                    {example, "rebar3 plugins list"},
+                    {short_desc, "List local and global plugins for this project"},
+                    {desc, "List local and global plugins for this project"},
+                    {opts, []}])),
+    {ok, State1}.
+
+-spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
+do(State) ->
+    GlobalConfigFile = rebar_dir:global_config(),
+    GlobalConfig = rebar_state:new(rebar_config:consult_file(GlobalConfigFile)),
+    GlobalPlugins = rebar_state:get(GlobalConfig, plugins, []),
+    GlobalPluginsDir = filename:join(rebar_dir:global_cache_dir(State), "plugins"),
+    display_plugins("Global plugins", GlobalPluginsDir, GlobalPlugins),
+
+    Plugins = rebar_state:get(State, plugins, []),
+    PluginsDir =rebar_dir:plugins_dir(State),
+    display_plugins("Local plugins", PluginsDir, Plugins),
+    {ok, State}.
+
+-spec format_error(any()) -> iolist().
+format_error(Reason) ->
+    io_lib:format("~p", [Reason]).
+
+display_plugins(_Header, _Dir, []) ->
+    ok;
+display_plugins(Header, Dir, Plugins) ->
+    ?CONSOLE("--- ~s ---", [Header]),
+    display_plugins(Dir, Plugins),
+    ?CONSOLE("", []).
```

Comment:
We should probably make sure we got similar output for all providers (help, plugins, deps, pkgs, etc.) I can start a task with a 3.0.0 milestone if you want.